### PR TITLE
Add TF BPs

### PR DESCRIPTION
**Pull request by ml4**
Add Terraform initial best practices document.

### CODE REVIEW

File: docs/terraform-module-best-practices.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Child modules are called in a root module configuration.
+A _root module_ is the top level Terraform configuration on which a `terraform plan` would be run.  It is defined [here](https://www.terraform.io/language/modules#the-root-module).
+A _terraform configuration_ generally refers to a whole configuration processed by Terraform during a run, but colloquially may be meant to refer to the root module.
+For Terraform Enterprise users, _submodules_ are also visible in [the documentation](https://www.terraform.io/enterprise/registry/using#viewing-nested-modules-and-examples) which are nested modules in subdirectories of configuration repositories.  These can also be structured in open source.
+
+## Best Practices
+This section details bullets which draw attention to good practices when handling Terraform deployments. Items referred in linked references are reproduced here to drive home specific best practices.
+
+- Read the documentation, which form the main best practices:
+  - Read the [Terraform module documentation]([Terraform Modules](https://www.terraform.io/language/modules)) before attempting to develop one's own child module.
```
Comment (eddie-knight) [inline docs/terraform-module-best-practices.md:19]: Should be:
```
  - Read the [Terraform module documentation](https://www.terraform.io/language/modules) before attempting to develop one's own child module.
```

**Review by eddie-knight — CHANGES_REQUESTED** [review]

**Comment by eddie-knight** [comment]
@ml4 Looks like you'll need to set up your CLA as an individual or corporate contributor (depending on your employer's policies)

**Comment by thinkl33t** [comment]
Should the first two sections in this be paragraphs or bulletpoints?

Because the file is markdown, right now the newlines are being stripped and it makes the paragraph a little hard to read.

**Comment by ml4** [comment]
Thanks @eddie-knight - weird though I set up my CLA contrib the other day.

**Comment by ml4** [comment]
@thinkl33t Thanks - added bullets to Preamble. I imagine most readers will be able to handle the introductory paragraph OK though. Let me know what you think.

**Review by thinkl33t — APPROVED** [review]
LGTM

**Review by eddie-knight — APPROVED** [review]

**Pull request merged**